### ISSUE TITLE
`pipeTo` with `preventCancel: true` never settle if `readable` doesn't produce new chunks

### ISSUE THREAD

**Issue by yume-chan** [MattiasBuelens/web-streams-polyfill#118]
### Repro

```js
const readable = new ReadableStream();
const abortController = new AbortController();

readable.pipeTo(new WritableStream(), {
    preventCancel: true,
    signal: abortController.signal,
}).then(() => {
    console.log('resolve');
}, () => {
    console.log('rejected');
});

setTimeout(() => {
    console.log('abort');
    abortController.abort();
}, 1000);

setTimeout(() => {
    console.log('exit');
}, 10000);
```

### Expected

Microsoft Edge canary 103.0.1255.0, with native `ReadableStream` and `WritableStream`

```text
abort
rejected
exit
```

Node.js v16.13.1, with `stream/web`

```text
(node:14660) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
abort
rejected
exit
```

### Actual

web-streams-polyfill: 4.0.0-beta.2

```text
abort
exit
```

**Comment by MattiasBuelens** [comment]
Good catch! Indeed, the `next` version of the polyfill seems to deviate from the specification. 🤔 

[The polyfill waits for both reads and writes to finish](https://github.com/MattiasBuelens/web-streams-polyfill/blob/v4.0.0-beta.2/src/lib/readable-stream/pipe.ts#L312-L313) before shutting down the pipe. However, [the reference implementation](https://github.com/whatwg/streams/blob/e9355ce79925947e8eb496563d599c329769d315/reference-implementation/lib/abstract-ops/readable-streams.js#L293-L294) and most browser implementations only wait until all *writes* have finished. Any read requests that have been started by the pipe but haven't finished yet are "silently" discarded.

I think I did this because we had to ensure there were no more pending read requests before the shutdown process calls `reader.releaseLock()` (see https://github.com/MattiasBuelens/web-streams-polyfill/commit/030bb2c716c4e4cb3d3e26a5ed275b4c721f3bfd). However, that was *before* we changed `releaseLock()`'s behavior in https://github.com/whatwg/streams/pull/1168. So really, the polyfill no longer needs to (and actually *shouldn't*) do this anymore.

I'll look into it. 🕵️

**Comment by MattiasBuelens** [comment]
Fixed in [v4.0.0-beta.3](https://github.com/MattiasBuelens/web-streams-polyfill/releases/tag/v4.0.0-beta.3)
